Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionMicrosoftSysmon.yaml

```diff
@@ -0,0 +1,170 @@
+        | project-away
+                       Provider,
+                       Channel,
+                       Task,
+                       Data,
+                       RawEventData,
+                       EventOriginId
+      };
+  union isfuzzy=true Sysmon3_Event,Sysmon3_WindowsEvent
+        | extend
+                SrcUsernameType = iff(not(Initiated), "Windows", ""),
+                SrcUsername = iff(not(Initiated), tostring(User), ""),
+                SrcProcessId = iff(not(Initiated), tostring(ProcessId), ""),
+                SrcProcessGuid = iff(not(Initiated), ProcessGuid, ""),
+                SrcProcessName = iff(not(Initiated), tostring(Image), ""),
+                SrcAppName = iff(not(Initiated), tostring(Image), ""),
+                SrcAppType = iff(not(Initiated), 'Process', ""),
+                DstUsernameType = iff(Initiated, "Windows", ""),
+                DstUsername = iff(Initiated, tostring(User), ""),
+                DstProcessId = iff(Initiated, tostring(ProcessId), ""),
+                DstProcessGuid = iff(Initiated, ProcessGuid, ""),
+                DstProcessName = iff(Initiated, tostring(Image), ""),
+                DstAppName = iff(Initiated, tostring(Image), ""),
+                DstAppType = iff(Initiated, 'Process', "")
+        | project-away  ProcessId, ProcessGuid, Image
+        | project-rename 
+                EventStartTime = UtcTime,
+                Dvc = Computer,
+                IpAddr = SourceIp,
+                Dst = DestinationIp,
+                DstPortNumber = DestinationPort,
+                SrcPortNumber = SourcePort,
+                NetworkRuleName = RuleName
+        | extend 
+                EventEndTime = EventStartTime,
+                Hostname = case(
+                                Initiated, DstHostname,
+                                not(Initiated), SrcHostname,
+                                Dvc),
+                Src = IpAddr,
+                DvcIpAddr = iff(Initiated, IpAddr, Dst),
+                SrcIpAddr = IpAddr,
+                DstIpAddr = Dst,
+                EventType = 'EndpointNetworkSession',
+                EventCount = int(1),
+                EventVendor = 'Microsoft',
+                EventSchemaVersion = '0.2.5',
+                EventSchema = 'NetworkSession', 
+                EventProduct = 'Sysmon',
+                EventResult = 'Success',
+                EventSeverity = 'Informational',
+                DvcOs = 'Windows',
+                Protocol = toupper(Protocol),
+                EventOriginalType = '3' // Set with a constant value to avoid parsing   
+        | extend
+                DvcHostname = Hostname
+        | extend
+                SrcHostname = iff( SrcHostname == "-", "", SrcHostname),
+                DvcHostname = iff( DvcHostname == "-", "", DvcHostname),
+                DstHostname = iff( DstHostname == "-", "", DstHostname) // let's make empty values actually empty
+        | invoke 
```

Comment:
This does not work. You cannot set SrcHostname (or Dvc or Dst) prior to calling the function. You will need to use a temp field (TempSrcHostname etc.), and then project away Temp*.